test/metabase/query_processor/streaming/xlsx_test.clj

```diff
@@ -451,7 +451,13 @@
            (second (xlsx-export [{:id 0, :name "Col"}] {} [["2020-03-28T10:12:06.681"]]))))
     (binding [qp.xlsx/*parse-temporal-string-values* true]
       (is (= [#inst "2020-03-28T10:12:06.681"]
-             (second (xlsx-export [{:id 0, :name "Col"}] {} [["2020-03-28T10:12:06.681"]]))))))
+             (second (xlsx-export [{:id 0, :name "Col" :effective_type :type/Temporal}]
+                                  {}
+                                  [["2020-03-28T10:12:06.681"]]))))
+      (is (= ["2020-03-28T10:12:06.681"]
```

Comment:
Suggestion: add the offending values from the original ticket to ensure that they work correctly, and a ticket number in the `testing` block:

```clojure
      (testing "Text values that are parseable as dates are not when effective_type is not temporal (#29708)"
        (doseq [value ["0001" "4161" "02" "2020-03-28T10:12:06.681"]]
          (is (= [value]
                 (second (xlsx-export [{:id 0, :name "Col" :effective_type :type/Text}]
                                      {}
                                      [[value]]))))))
```